x/claims/keeper/integration_test.go

```diff
@@ -0,0 +1,525 @@
+	initBalanceAmount0 := int64(math.Pow10(18) * 2)
+	initBalance0 := sdk.NewCoins(
+		sdk.NewCoin(stakeDenom, sdk.NewInt(delegationValue)),
+		sdk.NewCoin(claimsDenom, sdk.NewInt(initBalanceAmount0)),
+		sdk.NewCoin(evmtypes.DefaultEVMDenom, sdk.NewInt(initBalanceAmount0)),
+	)
+
+	var (
+		priv0              *ethsecp256k1.PrivKey
+		privs              []*ethsecp256k1.PrivKey
+		addr0              sdk.AccAddress
+		claimsRecords      []types.ClaimsRecord
+		params             types.Params
+		proposalId         uint64
+		totalClaimed       int64
+		remainderUnclaimed int64
+	)
+
+	BeforeAll(func() {
+		s.SetupTest()
+
+		params = s.app.ClaimsKeeper.GetParams(s.ctx)
+		params.EnableClaims = true
+		params.AirdropStartTime = s.ctx.BlockTime()
+		s.app.ClaimsKeeper.SetParams(s.ctx, params)
+
+		// mint coins for claiming and send them to the claims module
+		coins := sdk.NewCoins(sdk.NewCoin(claimsDenom, sdk.NewInt(totalClaimsAmount)))
+		err := s.app.BankKeeper.MintCoins(s.ctx, inflationtypes.ModuleName, coins)
+		s.Require().NoError(err)
+		err = s.app.BankKeeper.SendCoinsFromModuleToModule(s.ctx, inflationtypes.ModuleName, types.ModuleName, coins)
+		s.Require().NoError(err)
+
+		// fund testing accounts and create claim records
+		priv0, _ = ethsecp256k1.GenerateKey()
+		addr0 = getAddr(priv0)
+		testutil.FundAccount(s.app.BankKeeper, s.ctx, addr0, initBalance0)
+
+		for i := 0; i < accountCount; i++ {
+			priv, _ := ethsecp256k1.GenerateKey()
+			privs = append(privs, priv)
+			addr := getAddr(priv)
+			testutil.FundAccount(s.app.BankKeeper, s.ctx, addr, initBalance)
+			claimsRecord := types.NewClaimsRecord(sdk.NewInt(claimValue))
+			s.app.ClaimsKeeper.SetClaimsRecord(s.ctx, addr, claimsRecord)
+			acc := s.app.AccountKeeper.NewAccountWithAddress(s.ctx, addr)
+			s.app.AccountKeeper.SetAccount(s.ctx, acc)
+			claimsRecords = append(claimsRecords, claimsRecord)
+
+			balance := s.app.BankKeeper.GetBalance(s.ctx, addr, claimsDenom)
+			Expect(balance.Amount.Uint64()).To(Equal(uint64(initClaimsAmount)))
+		}
+
+		// ensure community pool doesn't have the fund
+		poolBalance := s.app.BankKeeper.GetBalance(s.ctx, distrAddr, claimsDenom)
+		Expect(poolBalance.Amount.Uint64()).To(Equal(uint64(0)))
+
+		// ensure module account has the escrow fund
+		balanceClaims := s.app.BankKeeper.GetBalance(s.ctx, claimsAddr, claimsDenom)
+		Expect(balanceClaims.Amount.Uint64()).To(Equal(uint64(totalClaimsAmount)))
+
+		s.Commit()
+
+		proposalId = govProposal(priv0)
+	})
+
+	Context("claiming before decay duration", func() {
+		var actionV int64
+		BeforeAll(func() {
+			actionV = actionValue
+		})
+
+		It("can claim ActionDelegate", func() {
+			addr := getAddr(privs[0])
+			prebalance := s.app.BankKeeper.GetBalance(s.ctx, addr, claimsDenom)
+			delegate(privs[0], 1)
+			balance := s.app.BankKeeper.GetBalance(s.ctx, addr, claimsDenom)
+			Expect(balance.Amount.Uint64()).To(Equal(uint64(actionV + int64(prebalance.Amount.Uint64()) - 1)))
+			totalClaimed += actionV
```

Comment:
I wanted to have a check on final balances, at the end of the tests. So, I used `totalClaimed` to keep track of the claimed amounts across leaf tests. But, I can also remove `totalClaimed += actionV` and test with the expected final balance values directly. 
Maybe this can be done better with `DescribeTable`, but I would leave this for a future PR.

In this context, I don't understand
> Would move `totalClaimed += actionV` to "did not clawback to the community pool"

Reply:
Removed the balance tracking from leaf tests https://github.com/tharsis/evmos/pull/373/commits/e8e68b79ac4faae7bf3d06d974bbc160d561c038